### PR TITLE
feat: new ebix response factory in incoming messages

### PR DESCRIPTION
**Pull request by MadsDue**
## Description

Grid operators (DDM), should be able to send metered data for a measurement point to EDI in the Ebix format, as described in the referenced task. Below is a list of the excepted PRs needed to prepare IncomingMessageClient to handle this new message

Previous PRs:
- [x] (IncomingMessageClient) Introduce a new incoming message type to represent the "metered data for a measurement point" message
- [x] (IncomingMessageClient) Implement MeteredDateForMeasurementPointEbixMessageParser
- [x] (IncomingMessageClient) Extend AuthorizeSender validator to ensure the sender role is MDR for this message type
- [x] (IncomingMessagePublisher) handle MeteredDataForMeasurementPoint 

This PR contains:
- [ ] (IncomingMessageClient) Implement an IResponseFactory to deliver an Ebix parsing error response.

Following PRs: 
- [ ] (IncomingMessageClient) Skip archiving message with inc doc type MeteredDataForMeasurementPoint

## Checklist
- [ ] Should the change be behind a feature flag?
- [ ] Can the feature be meaningfully disabled or circumvented if there are issues (e.g., database-breaking changes)?
- [x] Has it been considered whether data is being delivered to the wrong actor?
- [x] Subsystem test executed (dev_002/dev_003) =>  https://github.com/Energinet-DataHub/dh3-environments/actions/runs/11568752838
- [x] Is there time to monitor state of the release to Production?
- [x] Reference to the task => https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/351

### CODE REVIEW

File: source/IncomingMessages.Infrastructure/Response/EbixResponseFactory.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        var settings = new XmlWriterSettings() { OmitXmlDeclaration = true, };
+
+        using var writer = XmlWriter.Create(messageBody, settings);
+        writer.WriteStartElement("Error");
+        writer.WriteElementString("faultcode", "SOAP-ENV:Client");
+        writer.WriteElementString("faultstring", $"{result.Errors.First().Code}:{result.Errors.First().Message}");
+        writer.WriteStartElement("detail");
+        writer.WriteStartElement("fault");
+        writer.WriteElementString("ErrorCode", result.Errors.First().Code);
+        writer.WriteElementString("ErrorText", result.Errors.First().Message);
```
Comment (MWO1024) [inline source/IncomingMessages.Infrastructure/Response/EbixResponseFactory.cs:48]: Is there a chance we have more than one error? And if so, what should we do?

File: source/Tests/CimMessageAdapter/Response/EbixResponseFactoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    public void Generates_single_error_response()
+    {
+        var duplicateMessageIdError = new DuplicateMessageIdDetected("Duplicate message id");
+        var result = Result.Failure(duplicateMessageIdError);
+
+        var response = CreateResponse(result);
+
+        Assert.True(response.IsErrorResponse);
+        AssertHasValue(response, "faultcode", "SOAP-ENV:Client");
+        AssertHasValue(response, "faultstring", $"{duplicateMessageIdError.Code}:{duplicateMessageIdError.Message}");
```
Comment (MWO1024) [inline source/Tests/CimMessageAdapter/Response/EbixResponseFactoryTests.cs:45]: Would it make sense to have a verbatim comparison too? Mainly for a human reader, so it is easy to see what the response should look like?
```
<error>
<code>42</code>
<message>something went wrong</message>
</error>
```

**Review by MWO1024 — APPROVED** [review]

File: source/IncomingMessages.Infrastructure/Response/EbixResponseFactory.cs (hunk 1 above)
Comment (MadsDue) [reply source/IncomingMessages.Infrastructure/Response/EbixResponseFactory.cs:48]: Yes, there is. However, the ebix format can only handle 1 error at a time. The business has decided to always take the first error message.

File: source/Tests/CimMessageAdapter/Response/EbixResponseFactoryTests.cs (hunk 2 above)
Comment (MadsDue) [reply source/Tests/CimMessageAdapter/Response/EbixResponseFactoryTests.cs:45]: Currently I have followed the existing structure. But we could consider refactoring them all :) 

I have added a sub-task to the story with this.

**Pull request merged**